File: CSharpCodeBase/aiactions/scared.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class scared :aisequencenode{
 var AI_Scared = Class({__includes=AISequenceNode, init = public void (self, entity){
       var childNodes = {
         AI_ConditionalNode(entity, public void (entity){
             return #entity.relationship:GetTagTypes("scary") > 0;
         }),
         AI_GetTarget(entity),;
         AI_RunawayFromTarget(entity),;
       }
       AISequenceNode.init(self, entity, childNodes);
 }})
 return AI_Scared}}

File: CSharpCodeBase/aiactions/scared.lua
local Class = require 'utils.hump.class'
local AI_ConditionalNode = require 'utils.ailib.aiifnode'
local AISequenceNode = require 'utils.ailib.aisequence'
local AI_GetTarget = require 'aiactions.ai_gettarget'
local AI_RunawayFromTarget = require 'aiactions.ai_runawayfromtarget'

local AI_Scared = Class({__includes=AISequenceNode, init = function (self, entity)
      local childNodes = {
        AI_ConditionalNode(entity, function (entity)
            return #entity.relationship:GetTagTypes("scary") > 0
        end),
        AI_GetTarget(entity),
        AI_RunawayFromTarget(entity),
      }
      AISequenceNode.init(self, entity, childNodes)
end})

return AI_Scared